packages/visual-editor/src/components/VisualEditorRoot.tsx

```diff
@@ -60,5 +72,9 @@ export const VisualEditorRoot = ({ experience }: { experience?: Experience<Entit
 
   if (!initialized) return null;
 
+  if (isReadOnlyMode) {
+    return <ReadOnlyRenderer />;
+  }
+
```

Comment:
[4] Introducing a new Renderer that will handle elements diffently to the normal editor renderer